code/app/src/main/java/com/example/sigma_blue/TagList.java

```diff
@@ -0,0 +1,35 @@
+package com.example.sigma_blue;
+
+import java.util.ArrayList;
+
+/**
+ * Keeps track of a list of tags. Also allows for sync to the DB (well in the future...)
+ */
+public class TagList {
+    private ArrayList<Tag> tags;
+    final private TagDB tagDB;
+
+    public TagList(TagDB tagDB) {
+        this.tagDB = tagDB;
+    }
+
+    public void saveTagsToDB() {
+        tagDB.saveToDB(this);
+    }
+
+    public void refreshTagsFromDB() {
+        this.tags = tagDB.refreshFromDB();
+    }
+
+    public void addTag(Tag tag) {
+        tags.add(tag);
+    }
+
+    public boolean containsTag(Tag tag) {
```

Comment:
How is our handling of null input? Is that an exception thing? Are we trying to not throw any nulls at all? (I think eventually, nulls will get thrown, and we should do something about it.)

Reply:
Not sure. maybe we should talk about this later today